File: app/src/main/java/io/github/wulkanowy/ui/main/MainPresenter.java
package io.github.wulkanowy.ui.main;

import android.support.annotation.NonNull;

import javax.inject.Inject;

import io.github.wulkanowy.data.RepositoryContract;
import io.github.wulkanowy.ui.base.BasePresenter;

import static io.github.wulkanowy.utils.TimeUtilsKt.isHolidays;

public class MainPresenter extends BasePresenter<MainContract.View>
        implements MainContract.Presenter {

    private int fragmentCount = 0;

    @Inject
    MainPresenter(RepositoryContract repository) {
        super(repository);
    }

    @Override
    public void attachView(@NonNull MainContract.View view, int initTabId) {
        super.attachView(view);
        getView().showProgressBar(true);
        getView().hideActionBar();

        int tabPosition;

        if (initTabId != -1) {
            tabPosition = initTabId;
        } else {
            tabPosition = getRepository().getSharedRepo().getStartupTab();
        }

        getView().initiationBottomNav(tabPosition);
        getView().initiationViewPager(tabPosition);

        if (getRepository().getSharedRepo().isServicesEnable() && !isHolidays()) {
            getView().startSyncService(getRepository().getSharedRepo().getServicesInterval(),
                    getRepository().getSharedRepo().isMobileDisable());
        }
    }

    @Override
    public void onTabSelected(int position, boolean wasSelected) {
        if (!wasSelected) {
            getView().setCurrentPage(position);
        }
    }

    @Override
    public void onFragmentIsReady() {
        if (fragmentCount < 4) {
            fragmentCount++;
        }

        if (fragmentCount == 4) {
            getView().showActionBar();
            getView().showProgressBar(false);
        }
    }
}

File: app/src/main/java/io/github/wulkanowy/ui/main/MainPresenter.kt
package io.github.wulkanowy.ui.main

import io.github.wulkanowy.data.ErrorHandler
import io.github.wulkanowy.data.repositories.PreferencesRepository
import io.github.wulkanowy.ui.base.BasePresenter
import javax.inject.Inject

class MainPresenter @Inject constructor(
        errorHandler: ErrorHandler,
        private val prefRepository: PreferencesRepository)
    : BasePresenter<MainView>(errorHandler) {

    override fun onAttachView(view: MainView) {
        super.onAttachView(view)
        view.run {
            startMenuIndex = prefRepository.startMenuIndex
            initView()
        }
    }

    fun onViewStart() {
        view?.apply {
            currentViewTitle?.let { setViewTitle(it) }
            currentStackSize?.let {
                if (it > 1) showHomeArrow(true)
                else showHomeArrow(false)
            }
        }
    }

    fun onUpNavigate(): Boolean {
        view?.popView()
        return true
    }


    fun onBackPressed(default: () -> Unit) {
        view?.run {
            if (isRootView) default()
            else popView()
        }
    }

    fun onTabSelected(index: Int, wasSelected: Boolean): Boolean {
        return view?.run {
            if (wasSelected) {
                notifyMenuViewReselected()
                false
            } else {
                switchMenuView(index)
                true
            }
        } == true
    }
}
